dataland-frontend/src/components/resources/frameworkDataSearch/KpiDataObject.ts

```diff
@@ -4,6 +4,7 @@ export interface KpiDataObject {
   kpiKey: string;
   kpiLabel: string;
   kpiDescription: string;
+  kpiFormFieldComponent: string;
 
   [index: string]: string;
```

Comment:
I know it is working fine, but is this really a good idea in terms of readability?

Why don't we just give `KpiDataObject` a field `dataId` and a field `content` or `value` ?

Wouldn't this be much more understandble for an external reader of the code?

Reply:
Ok now I see.   Due to the way we build the map in the `LksgPanel.vue` it is not possible at the moment.
But it would be possible if a `KpiDataObject` would have a Map with dataId to value!
We could discuss this if you like.

Reply:
The main reason we construct this kind of objects, is because we need to provide an array of these objects to the `DataTable` component. I do not know if `DataTable` can handle a map. We would need to investigate.

Reply:
No you would still provide an array to `<DataTable>`.   But the objects in that array would have a field which contains a map of dataId to the value.

We can discuss this if you like.

Reply:
I have tried to change the field `[dataId]: kpiValue` to a map like `content: {[dataId]: kpiValue}` . This unfortunately breaks the table for more than one reporting period, the adding of data for more than one reportingPeriod to the table is handled with the `Object.assign()` method which is a bit tricky to handle. I will give this one to Benedikt to investigate.

Reply:
I managed to push a working version. Still Benedikt is the one who knows about this and can continue if something is broken now.